config/sources/families/tqma8mpxl.conf

```diff
@@ -0,0 +1,95 @@
+#
+# SPDX-License-Identifier: GPL-2.0
+#
+# Copyright (c) Martin Schmiedel
+#
+# This file is a part of the Armbian Build Framework
+# https://github.com/armbian/build/
+#
+
+declare -g ARCH="arm64"
+declare -g KERNEL_IMAGE_TYPE="Image"
+declare -g OFFSET="32"
+declare -g SERIALCON=ttymxc3
+declare -g IMX_FIRMWARE="firmware-imx-8.12"
+declare -g BOOTENV_FILE="tqma8mpxl.txt"
+declare -g ASOUND_STATE="asound.state.tqma8mpxl"
+
+# ATF related
+# MAP = target_make;target_patchdir;target_files
+declare -g ATF_TARGET_MAP="PLAT=imx8mp IMX_BOOT_UART_BASE=0x30a60000 bl31;;build/imx8mp/release/bl31.bin"
+# If needed use tq-systems repo
+ATFSOURCE='https://github.com/tq-systems/atf'
+ATFBRANCH="branch:TQMa8-imx_5.4.70_2.3.0"
+
+# Firmware related
+
+pre_config_uboot_target()
+{
+	# get the firmware
+	rm -rf ${IMX_FIRMWARE}*
+	wget https://www.nxp.com/lgfiles/NMG/MAD/YOCTO/${IMX_FIRMWARE}.bin -O ${IMX_FIRMWARE}.bin
+	chmod +x ${IMX_FIRMWARE}.bin
+	./${IMX_FIRMWARE}.bin --auto-accept
+
+	# copy needed files for uboot
+	cp ${IMX_FIRMWARE}/firmware/hdmi/cadence/signed_hdmi_imx8m.bin .
+	cp ${IMX_FIRMWARE}/firmware/ddr/synopsys/lpddr4*.bin .
+	export ATF_LOAD_ADDR=0x00970000
+}
+
+# Bootloader releated
+BOOTSOURCE='https://github.com/tq-systems/u-boot-tqmaxx.git'
+# MAP = target_make,target_patchdir,target_files
+# use xxx:xxx to rename files after build
+UBOOT_TARGET_MAP="flash.bin;;flash.bin"
+BOOTBRANCH='commit:81e390a605a4e69123dc7a9e3c58e886d461809e'
+# change the u-boot environment and config with patches
+BOOTPATCHDIR="u-boot-tq"
+#TODO: it could be a better way to use bootscript
+
+
+case $BRANCH in
+
+    current) # lts mainline kernel
+		KERNEL_MAJOR_MINOR="6.1"
+		KERNELBRANCH='tag:v6.1.71'
+		KERNELPATCHDIR='tqma8mpxl-current'
+	;;
+    edge|default) # recent mainline kernel
+		KERNEL_MAJOR_MINOR="6.6"
+		KERNELBRANCH='tag:v6.6.10'
```

Comment:
this is usually safe to attach to branch. we set fixed versions within a releases. If pure mainline kernel source is used and only few patches for this hw are added, its better to name kernel more generic, like imx8 ... this should work without a problem, while pure generic aarch64 could be troublesome.